kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/SharedKafkaProxyContext.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.kafkaproxyspec.Clusters;
+import io.kroxylicious.kubernetes.operator.config.RuntimeDecl;
+
+/**
+ * Encapsulates access to the mutable state in the {@link Context<KafkaProxy>} shared between
+ * {@link ProxyReconciler} its dependent resources.
+ */
+public class SharedKafkaProxyContext {
+
+    private SharedKafkaProxyContext() {
+    }
+
+    static final String RUNTIME_DECL_KEY = "runtime";
+    static final String ERROR_KEY = "error";
+
+    /**
+     * Set the RuntimeDecl
+     */
+    static void runtimeDecl(Context<KafkaProxy> context, RuntimeDecl runtimeDecl) {
+        context.managedDependentResourceContext().put(RUNTIME_DECL_KEY, runtimeDecl);
+    }
+
+    /**
+     * Get the RuntimeDecl
+     */
+    static RuntimeDecl runtimeDecl(Context<KafkaProxy> context) {
+        return context.managedDependentResourceContext().getMandatory(RUNTIME_DECL_KEY, RuntimeDecl.class);
+    }
+
+    /**
+     * Associate an Accepted condition with a specific cluster.
+     */
+    static void addClusterCondition(Context<KafkaProxy> context, Clusters cluster, ClusterCondition clusterCondition) {
+        Map<String, ClusterCondition> map = context.managedDependentResourceContext().get(ERROR_KEY, Map.class).orElse(null);
```

Comment:
nit: why is the key errors? I'd have thought we `conditions` was a more accurate key.

Reply:
Fixed